sui-execution/latest/sui-move-natives/src/crypto/nitro_attestation.rs

```diff
@@ -0,0 +1,148 @@
+        Ok((signature, signed_message, payload)) => {
+            let cert_chain_length = payload.get_cert_chain_length();
+            native_charge_gas_early_exit_option!(
+                context,
+                cost_params
+                    .verify_base_cost
+                    .and_then(|base_cost| cost_params
+                        .verify_cost_per_cert
+                        .map(|per_cert| base_cost + per_cert * (cert_chain_length as u64).into()))
+            );
+            match verify_nitro_attestation(&signature, &signed_message, &payload, current_timestamp)
+            {
+                Ok(()) => {
+                    // Encapsulate as a lambda and call to allow us to capture any `Err` returns.
+                    // Could do this with `and_then` as well if desired.
+                    let result = || {
+                        Ok(Value::struct_(Struct::pack(vec![
+                            Value::vector_u8(payload.module_id.as_bytes().to_vec()),
+                            Value::u64(payload.timestamp),
+                            Value::vector_u8(payload.digest.as_bytes().to_vec()),
+                            to_indexed_struct(payload.pcrs)?,
+                            to_option_vector_u8(payload.public_key)?,
+                            to_option_vector_u8(payload.user_data)?,
+                            to_option_vector_u8(payload.nonce)?,
+                        ])))
+                    };
+                    NativeResult::map_partial_vm_result_one(context.gas_used(), result())
+                }
+                Err(_) => Ok(NativeResult::err(context.gas_used(), VERIFY_ERROR)),
+            }
+        }
+        Err(_) => Ok(NativeResult::err(context.gas_used(), PARSE_ERROR)),
+    }
+}
+
+// Build an Option<vector<u8>> value
+fn to_option_vector_u8(value: Option<Vec<u8>>) -> PartialVMResult<Value> {
+    let vector_u8_type = Type::Vector(Box::new(Type::U8));
+    match value {
+        // Some(<vector<u8>>) = { vector[ <vector<u8>> ] }
+        Some(vec) => Ok(Value::struct_(Struct::pack(vec![Vector::pack(
+            &vector_u8_type,
+            vec![Value::vector_u8(vec)],
+        )?]))),
+        // None = { vector[ ] }
+        None => Ok(Value::struct_(Struct::pack(vec![Vector::empty(
+            &vector_u8_type,
+        )?]))),
+    }
+}
+
+// Convert a list of PCRs into a vector of PCREntry struct with index and value, 
+// where the indices are [0, 1, 2, 3, 4, 8] since AWS currently supports PCR0, 
+// PCR1, PCR2, PCR3, PCR4, PCR8.
+fn to_indexed_struct(pcrs: Vec<Vec<u8>>) -> PartialVMResult<Value> {
+    let indices = [0, 1, 2, 3, 4, 8];
+    let mut indexed_struct = vec![];
+    for (index, pcr) in pcrs.iter().enumerate() {
+        indexed_struct.push(Value::struct_(Struct::pack(vec![
+            Value::u8(indices[index]),
+            Value::vector_u8(pcr.to_vec()),
+        ])));
+    }
+    Ok(Value::vector_for_testing_only(indexed_struct))
+}
```

Comment:
Nit, as many editors and POSIX tools will add a newline
```suggestion
}
```